src/iterators.rs

```diff
@@ -641,3 +641,157 @@ pub fn new_axis_iter_mut<A, D>(v: ArrayViewMut<A, D>,
         life: PhantomData,
     }
 }
+
+/// An iterator that traverses over the specified axis
+/// and yields subviews of the specified size on this axis.
+///
+/// For example, in a 2 × 8 × 3 array, if the axis of iteration
+/// is 1 and the chunk size is 2, the yielded elements
+/// are 2 × 2 × 3 subviews (and there are 4 in total).
+///
+/// Iterator element type is `ArrayView<'a, A, D>`.
+pub struct ChunkIter<'a, A: 'a, D> {
+    iter: OuterIterCore<A, D>,
+    last_ptr: *mut A,
+    last_dim: D,
+    life: PhantomData<&'a A>,
+}
+
+fn chunk_iter_parts<A, D: Dimension>(v: ArrayView<A, D>,
+                                     axis: usize,
+                                     size: usize
+                                    ) -> (OuterIterCore<A, D>, *mut A, D)
```

Comment:
Ok, I'll try not to use that style. Btw maybe there should be a rustfmt file for the crate at some point?